apache/ozone
# HDDS-8017. Add check in ClosingContainerHandler for RM to handle Missing Containers

## What changes were proposed in this pull request?
Containers in `CLOSING` state are added to `MISSING` state as well.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-8017

## How was this patch tested?
- unit tests
- manual tests

Steps for manual tests:
- Set property `hdds.scm.replication.enable.legacy` to `false`
- Start ozone dev cluster with 3DNs
- Generate keys via `ozone freon rk --numOfKeys=10 --numOfBuckets=10 --type=RATIS --replication=ONE` command
- Check whether there are 3 containers in `OPEN` state using `ozone admin container report`
- Stop two DNs
- Check whether there are 1 container in `OPEN` state, 2 in `CLOSING` and 2 in `MISSING` state using `ozone admin container report`
